### PR TITLE
make `julia` optional

### LINKED ISSUE #151: make Julia optional dependency
- used python implementation if Julia isn’t installed but throw warning
- use xr_bitround instead of jl_bitround

- helps #95 and Julia installation issues 
- Continues #126 
- In the pangeo show case @rsignell-usgs also copied some code to not import xbitinfo due to julia

### PR DESCRIPTION
**Pull request by aaronspring**
closes #151

### CODE REVIEW

File: xbitinfo/julia_helpers.py [hunk 1]
```diff
@@ -17,7 +17,10 @@ def install(julia_project=None, quiet=False):  # pragma: no cover
 
     Also updates the local Julia registry.
     """
-    import julia
+    try:
+        import julia
+    except ImportError:
+        return True
```
Comment (aaronspring) [inline xbitinfo/julia_helpers.py:23]: quick fix and actually wrong. FIXME

**Review by observingClouds — APPROVED** [review]
With #156 being finally merged we can have a look here again. Because we haven't all Julia options implemented yet, we might want to hold this PR even further... on the other hand large datasets are currently not feasible with the wrapped Julia code and those are at the heart of why we developed this package in the first place. So we might just go ahead and make the python implementation the default and the Julia one optional (as suggested in the PR). Any feelings?

**Comment by aaronspring** [comment]
Would make xb pip installable. Then also python implementation should be default. But you are opening Pandora's box of having to reimplement everything from bitinformation.jl - those feature requests will come for sure

**Comment by ocefpaf** [comment]
> But you are opening Pandora's box of having to reimplement everything from bitinformation.jl - those feature requests will come for sure

Indeed. I know that you probably already know this so I don't want to sound like a broken record. Ideally, both the Julia and Python version should have a common core in a language that is easier to wrap, like C/C++ or Rust. Keeping two code bases aligned is a nightmare! But I also get that installing Julia to use `bitinformation.jl` is no picnic.

**Comment by observingClouds** [comment]
Thanks for the reference @ocefpaf. I think the conda installation is quite straight forward with all the great work you put into. The biggest obstacle is as you mention the wrapping. I think Julia is quite well in wrapping python code, but not the other way round yet.

**Pull request merged**